### PR TITLE
Add the `distill` subcommand

### PR DESCRIPTION
**Pull request by danieldk**
This is the first iteration of model distillation. The command uses a
soft loss to distill a model by default. A hard loss can be added
using the `hard-loss` option.

### CODE REVIEW

File: sticker-utils/src/subcommands/distill.rs [hunk 1]
```diff
@@ -0,0 +1,721 @@
...
+                let read_progress =
+                    ReadProgress::new(train_file.try_clone()?).or_exit("Cannot open train file", 1);
+
+                let progress_bar = read_progress.progress_bar().clone();
+                progress_bar
+                    .set_style(ProgressStyle::default_bar().template(
+                        "[Time: {elapsed_precise}, ETA: {eta_precise}] {bar} {percent}%",
+                    ));
+
+                let n_sentences = count_conllx_sentences(BufReader::new(read_progress))?;
```
Comment (twuebi) [inline sticker-utils/src/subcommands/distill.rs:710]: Is this full pass over the train data done to obtain a progress bar?

Comment (danieldk) [reply sticker-utils/src/subcommands/distill.rs:710]: We could get a progress bar using just the `Read + Seek` (as is done in finetune), though since we have it, it is also used for that.

The sentence count is done to count how many steps an epoch is, so that the user can specify `--epochs N`, since the main loop is step/batch-driven.

Comment (twuebi) [reply sticker-utils/src/subcommands/distill.rs:710]: Ah, wouldn't epoch-signalling also be possible by returning a None batch at the end of the file? I can also live with what it is now, the one-off read-through is probably dwarfed by the training time anyways. I guess it's a minor annoyance when working with really large files.

Comment (danieldk) [reply sticker-utils/src/subcommands/distill.rs:710]: It's really cheap. Usually < 1 minute, since it does not actually parse the CoNLL-X, but just counts blank lines.

**Review by twuebi — APPROVED** [review]
